nixos/modules/services/networking/hysteria.nix

```diff
@@ -0,0 +1,108 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.hysteria;
+in
+{
+  options.services.hysteria = {
+    enable = lib.mkEnableOption "Hysteria, a powerful, lightning fast and censorship resistant proxy";
+
+    package = lib.mkPackageOption pkgs "hysteria" { };
+
+    mode = lib.mkOption {
+      type = lib.types.enum [
+        "server"
+        "client"
+      ];
+      default = "server";
+      description = "Whether to use Hysteria as a client or a server.";
+    };
+
+    configFile = lib.mkOption {
+      default = null;
+      type = lib.types.nullOr lib.types.path;
```

Comment:
wouldn't it be better to add a `settings` or `config` submodule where you can set the config declaratively?

Reply:
We discussed it in #nixos-zhcn:nichi.co, `utils.genJqSecretsReplacementsSnippet` currently only supports a string as value, which is very convenient in our use case.

Reply:
Could you please add settings besides configFile?

Reply:
@linsui can you please review this patch? It's currently in my own flake and I'm not very confident about the change.

https://pb.ny4.dev/810B6a.patch

Reply:
Does hysteria also work with json?

Reply:
Yes, JSON YAML TOML

Reply:
I don't know if the preStart part necessary. Otherwise it looks good to me.